src/Microsoft.AspNet.Mvc.Core/ModelBinding/ArrayModelBinder.cs

```diff
@@ -25,15 +26,16 @@ public override Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingCo
             return base.BindModelAsync(bindingContext);
         }
 
-        protected override object CreateEmptyCollection()
+        /// <inheritdoc />
+        public override object CreateEmptyCollection(Type targetType)
         {
             return new TElement[0];
         }
 
         /// <inheritdoc />
-        protected override object GetModel(IEnumerable<TElement> newCollection)
+        protected override object GetModel(Type targetType, IEnumerable<TElement> collection)
         {
-            return newCollection?.ToArray();
+            return collection?.ToArray();
```

Comment:
Will the default case here be that the model is already an array? What are the cases we might get here and have it not be an array?